src/cpp/core/include/core/Algorithm.hpp

```diff
@@ -200,7 +201,23 @@ void append(ContainerType* pContainer, const ContainerType& other)
 inline std::vector<std::string> split(const std::string& string, const std::string& delim)
 {
    std::vector<std::string> result;
-   boost::algorithm::split(result, string, boost::is_any_of(delim));
+   
```

Comment:
As far as I can see, `boost::algorithm::split()` only allows single-character delimiters, and so passing e.g. `||` here would actually just attempting to split on `|`. This is of course clear when attempting to call `boost::algorithm::split` directly (since you need to use the `is_any_of()` helper function) but is masked when using this helper `core::algorithm::split()` function.

In all previous cases, the usage of `core::algorithm::split()` was with a single character (`"\n"`) so I think this change is safe + correct.